routes/stake.go

```diff
@@ -405,13 +431,257 @@ func (fes *APIServer) UnlockStake(ww http.ResponseWriter, req *http.Request) {
+		_AddInternalServerError(ww, "GetStakesForValidator: Problem encoding response as JSON")
+		return
+	}
+}
+
+// GetLockedStakesForValidatorAndStaker returns all locked stake entries for a given validator and staker
+// If lockedAtEpochNumber is specified, only the locked stake entry that was locked at that epoch number is returned
+// If startEpochNumber and endEpochNumber are specified, all locked stake entries that were locked between those.
+// If none are provided, all locked stake entries are returned.
+func (fes *APIServer) GetLockedStakesForValidatorAndStaker(ww http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	validatorPublicKeyBase58Check, validatorExists := vars["validatorPublicKeyBase58Check"]
+	if !validatorExists {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: validatorPublicKeyBase58Check is required"))
+		return
+	}
+	stakerPublicKeyBase58Check, stakerExists := vars["stakerPublicKeyBase58Check"]
+	if !stakerExists {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: stakerPublicKeyBase58Check is required"))
+		return
+	}
+
+	// Create UTXO View
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: Problem fetching utxoView: %v", err))
+		return
+	}
+
+	// Convert validator public key to bytes
+	validatorPKID, err := fes.getPKIDFromPublicKeyBase58Check(utxoView, validatorPublicKeyBase58Check)
+	if err != nil || validatorPKID == nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: Problem decoding validator public key: %v", err))
+		return
+	}
 
-// Other functions to implement.
-// 1. _convertStakeEntryToResponse() helper function to convert a StakeEntry to a StakeEntryResponse.
-// 2. _convertLockedStakeEntryToResponse() helper function to convert a LockedStakeEntry to a LockedStakeEntryResponse.
+	// Convert staker public key to bytes
+	stakerPKID, err := fes.getPKIDFromPublicKeyBase58Check(utxoView, stakerPublicKeyBase58Check)
+	if err != nil || stakerPKID == nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: Problem decoding staker public key: %v", err))
+		return
+	}
+
+	queryParamBytes, err := json.Marshal(req.URL.Query())
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: Problem parsing query params: %v", err))
+		return
+	}
+
+	queryParams := make(map[string][]string)
+
+	if err = json.Unmarshal(queryParamBytes, &queryParams); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetLockedStakesForValidatorAndStaker: Problem parsing query params: %v", err))
+		return
+	}
+
+	var lockedStakeEntries []*lib.LockedStakeEntry
+	// First check for lockedAtEpochNumber
+	if len(queryParams["lockedAtEpochNumber"]) != 0 {
```

Comment:
Please create consts for the query param keys: `lockedAtEpochNumber`, `startEpochNumber`, and `endEpochNumber`